Ecal/src/Ecal/EcalVetoProcessor.cxx

```diff
@@ -1261,6 +1279,44 @@ float EcalVetoProcessor::distPtToLine(TVector3 h1, TVector3 p1, TVector3 p2) {
   return ((h1 - p1).Cross(h1 - p2)).Mag() / (p1 - p2).Mag();
 }
 
+std::vector<float> EcalVetoProcessor::trackProp(const ldmx::Tracks &tracks,
+                                                ldmx::TrackStateType ts_type,
+                                                const std::string &ts_title) {
+  // Vector to hold the new track state variables
+  std::vector<float> newTrackStates;
+
+  // Return if no tracks
+  if (tracks.empty()) return newTrackStates;
+
+  // Otherwise loop on the tracks
+  for (auto &track : tracks) {
+    // Get track state for ts_type
+    auto trk_ts = track.getTrackState(ts_type);
+    // Continue if there's no value
+    if (!trk_ts.has_value()) continue;
+    ldmx::Track::TrackState &TargetState = trk_ts.value();
```

Comment:
This should actually be 
```suggestion
    ldmx::Track::TrackState &EcalTrackState = trk_ts.value();
```

Reply:
Also, please have the variable be `snake_case` instead of `CamelCase` so that it is easy to visually distinguish from classes (`CamelCase`) and functions (`someWeirdJavaThing`).